### PR TITLE
Compare version with appVersion, version fields in catalog index

### PR DESCRIPTION
**Pull request by tomahawk28**
In case we need to show the upstream version as appVersion, we should compare appVersion with `version` in appcatalog index files.

### CODE REVIEW

File: appcatalog.go [hunk 1]
```diff
@@ -63,7 +63,15 @@ func getLatestEntry(ctx context.Context, storageURL, app, appVersion string) (en
...
-		if appVersion != "" && !strings.HasSuffix(e.AppVersion, appVersion) {
+		if appVersion == "" {
+			continue
+		}
+
+		if !strings.HasSuffix(e.AppVersion, appVersion) {
+			continue
+		}
+
+		if !strings.HasSuffix(e.Version, appVersion) {
```
Comment (tomahawk28) [inline appcatalog.go:74]: this would fall back to the version comparison.

Comment (MarcelMue) [reply appcatalog.go:74]: Okay so the case would be that neither version has the suffix, right? I don't quite understand how this constructs the correct version if only given the SHA

**Review by MarcelMue — APPROVED** [review]
Might be okay but I am not sure exactly how the index stuff works.

File: appcatalog.go (hunk 1 above)
Comment (tomahawk28) [reply appcatalog.go:74]: I think it's a bit hard to get this part since you need to understand how we get the entry from [the catalog entry](https://github.com/giantswarm/default-test-catalog/blob/master/index.yaml). This function would gather all entries with matching SHA with `appVersion` and `version` and later return the latest entry. (I see one problem with that appVersion is now SHA string in apptest, though)  This is nothing to do with a construct with SHA actually.

Comment (rossf7) [reply appcatalog.go:74]: What about adding comments explaining the 2 cases?

It makes sense if you're familiar with the index.yaml format but otherwise its a bit hard to follow.

**Review by rossf7 — APPROVED** [review]
LGTM but I think some comments on the 2 cases would help.
